File: Software Development Module/Revision/Exam2016/PasswordGenerator.java
/*
PasswordGenerator.java
Billy Cussen
11/08/2020
*/

public class PasswordGenerator{

    //Data Members
    private String input;
    private String output;
    private final String VOWELS = "aieou";
    private StringBuffer sbOutput;

    //Constructor
    public PasswordGenerator(){
        this.input = new String();
        this.output = new String();
        this.sbOutput = new StringBuffer();
    }

    //Compute, Setter & Getter
    public String getPassword(String input){
        this.input = input.toLowerCase();
        this.sbOutput.append(this.input);
        this.sbOutput.reverse();

        for(int k = 0; k < this.sbOutput.length(); k++){

            for(int j = 0; j < this.VOWELS.length(); j++){

                if(this.sbOutput.charAt(k) == this.VOWELS.charAt(j)){
                    this.sbOutput.replace(k,k+1,this.input.length()+"");
                    break;
                } 
                
            }

            if(this.sbOutput.charAt(k) == ' '){
                this.sbOutput.replace(k, k+1,"+");
            } 

        }
        this.output = this.sbOutput.toString();
        this.sbOutput = null;

        return this.output;
    }

}

File: Python/Software Development Module - Python/Revision/Exam2016/PasswordGenerator/PasswordGenerator.py
"""
PasswordGenerator.py
Billy Cussen
31/10/2020
"""

class myPasswordGenerator:

    def __init__(self, fullName):
        self.VOWELS = "aeiou"
        self.fullName = fullName

    def getPassword(self):
        self.fullName = self.fullName.lower()
        password = ""
        for c in self.fullName:
            password = c + password

        for c1 in self.fullName:
            if c1 in self.VOWELS:
                password = password.replace(c1,"")
            
        for c2 in password:
            if c2 == ' ':
                    password = password.replace(" ","+")

        password = password.title()
        return password